src/Microsoft.AspNet.Mvc.Xml/XmlDataContractSerializerOutputFormatter.cs

```diff
@@ -183,11 +185,31 @@ public override Task WriteResponseBodyAsync([NotNull] OutputFormatterContext con
                     obj = wrapperProvider.Wrap(obj);
                 }
 
-                var dataContractSerializer = CreateSerializer(wrappingType);
+                var dataContractSerializer = GetCachedSerializer(wrappingType);
                 dataContractSerializer.WriteObject(xmlWriter, obj);
             }
 
             return Task.FromResult(true);
         }
+
+        private DataContractSerializer GetCachedSerializer(Type type)
+        {
+            if (type == null)
```

Comment:
also if a test is passing null for `Type` here, not sure how realistic it is..you should modify the test instead i think...